src/main/java/org/jabref/gui/preferences/TableColumnsTab.java

```diff
@@ -96,43 +92,50 @@
     public TableColumnsTab(JabRefPreferences prefs, JabRefFrame frame) {
         this.prefs = prefs;
         this.frame = frame;
-        this.data = FXCollections.observableArrayList(
-                new TableRow("entrytype",75),
-                new TableRow("author/editor",300),
-                new TableRow("title",470),
-                new TableRow("year",60),
-                new TableRow("journal",130),
-                new TableRow("bibtexkey",100));
 
+        /* Populate the data of Entry table columns */
+        List<String> prefColNames = this.prefs.getStringList(this.prefs.COLUMN_NAMES);
+        this.data = FXCollections.observableArrayList();
+        for (int i = 0; i < prefColNames.size(); i++) {
+            this.data.add(new TableRow(prefColNames.get(i)));
+        }
+
+        /* UI for Entry table columns */
         colSetup = new TableView<>();
-        TableColumn<TableRow,String> field = new TableColumn<>(Localization.lang("Field name"));
-        TableColumn<TableRow,Double> column = new TableColumn<>(Localization.lang("Column width"));
+        TableColumn<TableRow, String> field = new TableColumn<>(Localization.lang("Field name"));
+        //TableColumn<TableRow,Double> column = new TableColumn<>(Localization.lang("Column width"));
```

Comment:
Please remove the commented out code That makes it easier to spot the actual changes and with version control you always have the full history of commits, so no need to comment out stuff